addon/utils/validation-errors.js

```diff
@@ -37,5 +41,12 @@ export default function buildMessage(key, result) {
     return messages.formatMessage(message, assign({ description }, context));
   }
 
-  return messages.formatMessage(get(messages, type), assign({ description }, context));
+  let message = get(messages, type);
+  if (returnsRaw) {
+    context = assign({}, context, { description })
+    return { value, type, message, context };
```

Comment:
Possible to move this logic up ~ line 25?

Reply:
Ha ha, I've spent quite some time trying to figure out whether to do it or not before submitting the PR!

Here's my train of thoughts: the result.message and context.message are "exceptions" in a way, in other words opportunities given by the addon to circumvent the standard way of doing things.
I see my PR as "an alternative standard way of doing things", and as such, these properties can still make a lot of sense. I took the path of least resistance and implemented the PR so that it does not short circuit the behavior coming with the presence of *.message properties. TBD...

Reply:
All valid concerns. Thx for the explanation.